sympy/physics/vector/frame.py

```diff
@@ -476,71 +476,63 @@ def dcm(self, otherframe):
         otherframe._dcm_cache[self] = outdcm.T
         return outdcm
 
-    def orient(self, parent, rot_type, amounts, rot_order=''):
-        """Sets the orientation of this reference frame relative to another
-        (parent) reference frame.
+    def _dcm(self, parent, parent_orient):
+        # Reset the _dcm_cache of this frame, and remove it from the
+        # _dcm_caches of the frames it is linked to. Also remove it from the
+        # _dcm_dict of its parent
+        frames = self._dcm_cache.keys()
+        dcm_dict_del = []
+        dcm_cache_del = []
+        for frame in frames:
+            if frame in self._dcm_dict:
+                dcm_dict_del += [frame]
+            dcm_cache_del += [frame]
+        for frame in dcm_dict_del:
+            del frame._dcm_dict[self]
+        for frame in dcm_cache_del:
+            del frame._dcm_cache[self]
+        # Add the dcm relationship to _dcm_dict
+        self._dcm_dict = self._dlist[0] = {}
+        self._dcm_dict.update({parent: parent_orient.T})
+        parent._dcm_dict.update({self: parent_orient})
+        # Also update the dcm cache after resetting it
+        self._dcm_cache = {}
+        self._dcm_cache.update({parent: parent_orient.T})
+        parent._dcm_cache.update({self: parent_orient})
+
+    def orient_axis(self, parent, amounts):
```

Comment:
```suggestion
    def orient_axis(self, parent, axis, amount):
```

Rotation about an axis requires 3 three things: 1) parent frame to rotate relative to, 2) vector (axis) to rotate about, and 3) a scalar amount of rotation.

The api should be:

```
def orient_axis(self, reference_frame, vector, amount):
```

or

```
def orient_axis(self, parent_frame, vector_axis, amount):
```

It would be nice if the argument variable names were informative.